vector/src/main/scala/geotrellis/vector/io/shape/reader/MultiPolygonReader.scala

```diff
@@ -0,0 +1,161 @@
+package geotrellis.vector.io.shape.reader
+
+import geotrellis.vector._
+import com.vividsolutions.jts.{geom => jts}
+import com.vividsolutions.jts.algorithm.CGAlgorithms
+import spire.syntax.cfor._
+
+import scala.collection.mutable
+import java.nio.ByteBuffer
+
+import GeomFactory._
+
+object MultiPolygonReader {
```

Comment:
This code executed 100's of milliseconds faster than the other code. Probably due to wrapper object creation?